Makefile

```diff
@@ -306,13 +309,25 @@ INTEGRATION_TEST_TIMEOUT ?= "30m"
 CONFORMANCE_TEST_TIMEOUT ?= "20m"
 E2E_TEST_TIMEOUT ?= "20m"
 
+SETUP_ENVTEST_VERSION ?= 0.19.0
+SETUP_ENVTEST = $(PROJECT_DIR)/bin/installs/setup-envtest/$(SETUP_ENVTEST_VERSION)/bin/setup-envtest
+.PHONY: setup-envtest
+setup-envtest: mise ## Download setup-envtest locally if necessary.
+	@$(MAKE) mise-plugin-install DEP=setup-envtest URL=https://github.com/pmalek/mise-setup-envtest.git
+	@$(MISE) install setup-envtest@$(SETUP_ENVTEST_VERSION)
+
+.PHONY: use-setup-envtest
+use-setup-envtest:
+	$(SETUP_ENVTEST) use
+
 .PHONY: test
 test: test.unit
 
 UNIT_TEST_PATHS := ./controller/... ./internal/... ./pkg/... ./modules/...
 
 .PHONY: _test.unit
-_test.unit: gotestsum
+_test.unit: gotestsum setup-envtest use-setup-envtest
```

Comment:
I'd make the envtest based suite a separate one just like we have it for KIC: https://github.com/Kong/kubernetes-ingress-controller/blob/65ed7e953f08bc11ac7e8cecfe55cfb2b59b4219/Makefile#L428-L458.

The fact that we still have `envtest` tag in `_test.unit` (link: https://github.com/Kong/kubernetes-ingress-controller/blob/65ed7e953f08bc11ac7e8cecfe55cfb2b59b4219/Makefile#L407-L408) is a (harmless) mistake because we don't have any UTs that require the envtest harness.

Reply:
So do we want to move the reconciler tests to `envtest` suite? Creating reconcilers (Now `KongPluginBinding`) requires a k8s APIserver so they need a `envtest` environment.

Reply:
Yes, I'd say so.

Reply:
Added a dedicated `envtest` suite.